confgenerator/confgenerator_test.go

```diff
@@ -181,19 +181,25 @@ func updateOrCompareGolden(t *testing.T, testName string, goos string, expectedB
 	expected := strings.ReplaceAll(string(expectedBytes), "\r\n", "\n")
 	actual = strings.ReplaceAll(actual, "\r\n", "\n")
 	goldenPath := fmt.Sprintf(path, goos, testName)
-	if diff := cmp.Diff(expected, actual); diff != "" {
-		if *updateGolden {
+	diff := cmp.Diff(expected, actual)
+	if *updateGolden {
+		if diff != "" || !fileExists(goldenPath) {
```

Comment:
It's interesting that `readFileContent` assumes that any error from `ioutil.ReadFile` would be a "file not found" error. This means that if a non-empty file exists, but there's an intermittent error actually reading that file, the content will be assumed empty, and your code will not try to overwrite it, which would cause the tests to fail later (exactly what `-update_golden` is intended to prevent). If we want to catch that case, we should unconditionally attempt to overwrite the golden file when the content is empty (which will at least log a failure message in case of an intermittent error). Alternatively, we can fix line 168 to actually distinguish the various errors and only swallow the "file not found" ones… Does this make sense?